internal/engine/uisession/reducers.go

```diff
@@ -1,22 +1,25 @@
 package uisession
 
 import (
-	"context"
-
 	"github.com/tilt-dev/tilt/internal/store"
 	"github.com/tilt-dev/tilt/pkg/apis"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
 )
 
-func HandleUISessionCreateAction(_ context.Context, state *store.EngineState, a UISessionCreateAction) {
+func HandleUISessionCreateAction(state *store.EngineState, a UISessionCreateAction) {
 	key := apis.Key(a.UISession)
 	if _, ok := state.UISessions[key]; !ok {
 		state.UISessions[key] = a.UISession
 	}
 }
 
-func HandleUISessionUpdateStatusAction(ctx context.Context, state *store.EngineState, a UISessionUpdateStatusAction) {
+func HandleUISessionUpdateStatusAction(state *store.EngineState, a UISessionUpdateStatusAction) {
 	key := apis.KeyFromMeta(*a.ObjectMeta)
-	if _, ok := state.UISessions[key]; ok {
-		state.UISessions[key].Status = *a.Status
+	if orig, ok := state.UISessions[key]; ok {
+		state.UISessions[key] = &v1alpha1.UISession{
```

Comment:
also wan to call out this change to @milas . for the session stuff (where we're copying stuff to websockets), it's much easier of these are truly immutable, so we can copy without locking

Go generics can't come soon enough!!! not happy to have all this duplicated code